### PR TITLE
[Core/Feature]: Allow GMs to control accepting/declining party invites.

### PR DESCRIPTION
**Pull request by lineagedr**
## Changes Proposed:
-  Add a way for GMs to control accepting/declining party invites.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/7489

## Tests Performed:

- Tested with the method below

## How to Test the Changes:

1. Login with a GM character
2. Use the command .invites on/off
3. Try to invite your gm char

## Known Issues and TODO List:

- None so far

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
Does this save for characters if they logout or server restarts etc?

**Comment by lineagedr** [comment]
@Kitzunu 
doh

File: src/server/game/Handlers/GroupHandler.cpp [hunk 1]
```diff
@@ -81,6 +81,11 @@ void WorldSession::HandleGroupInviteOpcode(WorldPacket& recvData)
         return;
     }
 
+    if (GetPlayer()->IsGameMaster() && !GetPlayer()->IsAcceptingInvites())
```
Comment (Kitzunu) [inline src/server/game/Handlers/GroupHandler.cpp:84]: You could add this to the check above to also sendERR_INVITE_RESTRICTED

**Comment by Kitzunu** [comment]
> @Kitzunu
> doh

I know it wont :D

I was being subtle to suggest you add a playerflag for it like whisper gmmode etc

**Comment by lineagedr** [comment]
@Kitzunu It's WIP anyway

**Comment by Kitzunu** [comment]
Sure, just suggestion

File: src/server/game/Handlers/GroupHandler.cpp (hunk 1 above)
Comment (lineagedr) [reply src/server/game/Handlers/GroupHandler.cpp:84]: @Kitzunu The ERR_INVITE_RESTRICTED message is

```
Trial accounts cannot invite characters into groups.
```

Comment (lineagedr) [reply src/server/game/Handlers/GroupHandler.cpp]: ERR_BAD_PLAYER_NAME_S Should be fine

**Comment by Bogir** [comment]
variant as TOGGLE:
```
static bool HandleGMToggleInvitesModeCommand(ChatHandler* handler, char const* /*args*/)
    {
        if (sWorld->getBoolConfig(CONFIG_ALLOW_GM_GROUP) == true)
        {
            sWorld->setBoolConfig(CONFIG_ALLOW_GM_GROUP, false);
            handler->SendSysMessage("Now players can not invite you to their group");
        }
        else
        {
            sWorld->setBoolConfig(CONFIG_ALLOW_GM_GROUP, true);
            handler->SendSysMessage("Now players can not invite you to their group");
        }
        return true;
    }
    `

**Comment by lineagedr** [comment]
> variant as TOGGLE:
> 
> ```
> static bool HandleGMToggleInvitesModeCommand(ChatHandler* handler, char const* /*args*/)
>     {
>         if (sWorld->getBoolConfig(CONFIG_ALLOW_GM_GROUP) == true)
>         {
>             sWorld->setBoolConfig(CONFIG_ALLOW_GM_GROUP, false);
>             handler->SendSysMessage("Now players can not invite you to their group");
>         }
>         else
>         {
>             sWorld->setBoolConfig(CONFIG_ALLOW_GM_GROUP, true);
>             handler->SendSysMessage("Now players can not invite you to their group");
>         }
>         return true;
>     }
>     `
> ```

This looks like it will be for all GMs.

**Comment by Bogir** [comment]
> 
> @Bogir This looks like it will be for all GMs.

Yes, it just in my case, I am 1 GM :D
I suggest to use toggle and not use on/off

File: src/server/game/Entities/Player/Player.h [hunk 2]
```diff
@@ -517,6 +517,7 @@ enum PlayerExtraFlags
     PLAYER_EXTRA_SPECTATOR_ON       = 0x0080,               // Marks if player is spectactor
     PLAYER_EXTRA_PVP_DEATH          = 0x0100,               // store PvP death status until corpse creating.
     PLAYER_EXTRA_SHOW_DK_PET        = 0x0400,               // Marks if player should see ghoul on login screen
+    PLAYER_EXTRA_ACCEPT_INVITES     = 0x0800
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/Player.h:520]:
```suggestion
    PLAYER_EXTRA_ACCEPT_INVITES     = 0x0800               // Marks if player accepts whisper
```

Comment (FrancescoBorzi) [reply src/server/game/Entities/Player/Player.h:520]: whisper?

**Comment by FrancescoBorzi** [comment]
nice PR 👍

File: src/server/game/Entities/Player/Player.h (hunk 2 above)
Comment (Kitzunu) [reply src/server/game/Entities/Player/Player.h:520]: lol

**Comment by Kitzunu** [comment]
https://github.com/azerothcore/azerothcore-wotlk/pull/7490/commits/2f0aecc70c7dbbcac34fde9c538cc5392f96f29f kek

**Review by Kitzunu — APPROVED** [review]

File: src/server/game/Miscellaneous/Language.h [hunk 3]
```diff
@@ -1355,7 +1355,11 @@ enum AcoreStrings
...
 
     LANG_DEBUG_BG_CONF                            = 30085,
-    LANG_DEBUG_ARENA_CONF                         = 30086
+    LANG_DEBUG_ARENA_CONF                         = 30086,
+
+    LANG_COMMAND_INVITE_ACCEPTING                 = 30087,
+    LANG_COMMAND_INVITE_ACCEPT_ON                 = 30088,
+    LANG_COMMAND_INVITE_ACCEPT_OFF                = 30089,
 
     // 30087-30095 reserved for passive anticheat
```
Comment (Kitzunu) [inline src/server/game/Miscellaneous/Language.h:1364]: According to comment below these are reserved for anticheat

// 30087-30095 reserved for passive anticheat

**Comment by Kitzunu** [comment]
oh also dont forget to add the command in the `command` table

File: src/server/game/Entities/Player/Player.h [hunk 4]
```diff
@@ -509,15 +509,15 @@ enum PlayerExtraFlags
...
+    PLAYER_EXTRA_ACCEPT_WHISPERS    = 0x0004,               // Marks if GM accepts whispers
     PLAYER_EXTRA_TAXICHEAT          = 0x0008,
     PLAYER_EXTRA_GM_INVISIBLE       = 0x0010,
     PLAYER_EXTRA_GM_CHAT            = 0x0020,               // Show GM badge in chat messages
     PLAYER_EXTRA_HAS_310_FLYER      = 0x0040,               // Marks if player already has 310% speed flying mount
     PLAYER_EXTRA_SPECTATOR_ON       = 0x0080,               // Marks if player is spectactor
     PLAYER_EXTRA_PVP_DEATH          = 0x0100,               // store PvP death status until corpse creating.
     PLAYER_EXTRA_SHOW_DK_PET        = 0x0400,               // Marks if player should see ghoul on login screen
-    PLAYER_EXTRA_ACCEPT_INVITES     = 0x0800
+    PLAYER_EXTRA_ACCEPT_INVITES     = 0x0800                // Marks if GM accepts whispers
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Player/Player.h]: invites* ?

Comment (lineagedr) [reply src/server/game/Entities/Player/Player.h]: @FrancescoBorzi That's outdated.

File: src/server/game/Miscellaneous/Language.h (hunk 3 above)
Comment (Nefertumm) [reply src/server/game/Miscellaneous/Language.h:1364]: Kitzunu is right, please change this numbers :D

File: data/sql/updates/pending_db_world/rev_1629575483816529000.sql [hunk 5]
```diff
@@ -0,0 +1,10 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1629575483816529000');
+
+DELETE FROM `acore_string` WHERE `entry` IN (30087, 30088, 30089);
+
+INSERT INTO `acore_string` 
+    (`entry`, `content_default`, `locale_koKR`, `locale_frFR`, `locale_deDE`, `locale_zhCN`, `locale_zhTW`, `locale_esES`, `locale_esMX`, `locale_ruRU`) 
+VALUES
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1629575483816529000.sql:7]: Make the 3 lines above just 1, and then fix the indentation of the data. 👍

File: src/server/game/Miscellaneous/Language.h [hunk 6]
```diff
@@ -1355,7 +1355,11 @@ enum AcoreStrings
     LANG_BG_READY_CHECK_ERROR                     = 30084,
 
     LANG_DEBUG_BG_CONF                            = 30085,
-    LANG_DEBUG_ARENA_CONF                         = 30086
+    LANG_DEBUG_ARENA_CONF                         = 30086,
+
+    LANG_COMMAND_INVITE_ACCEPTING                 = 30087,
+    LANG_COMMAND_INVITE_ACCEPT_ON                 = 30088,
+    LANG_COMMAND_INVITE_ACCEPT_OFF                = 30089,
```
Comment (Nefertumm) [inline src/server/game/Miscellaneous/Language.h:1362]: Change these numbers, due from 30087 to 30095 is reserved.

File: data/sql/updates/pending_db_world/rev_1629575483816529000.sql [hunk 7]
```diff
@@ -0,0 +1,10 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1629575483816529000');
+
+DELETE FROM `acore_string` WHERE `entry` IN (30087, 30088, 30089);
+
+INSERT INTO `acore_string` 
+    (`entry`, `content_default`, `locale_koKR`, `locale_frFR`, `locale_deDE`, `locale_zhCN`, `locale_zhTW`, `locale_esES`, `locale_esMX`, `locale_ruRU`) 
+VALUES 
+    (30087,'Accepting Invites: %s',NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL),
+    (30088,'Accepting Invites: ON',NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL),
+    (30089,'Accepting Invites: OFF',NULL,NULL,NULL,NULL,NULL,NULL,NULL,NULL);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1629575483816529000.sql:10]: Insert the command to DB also? <3

**Review by Nefertumm — CHANGES_REQUESTED** [review]

**Comment by Nefertumm** [comment]
Please solve merge conflicts

**Comment by FrancescoBorzi** [comment]
> Please solve merge conflicts

@lineagedr

**Comment by Bogir** [comment]
This command has no parameters. Therefore it was nice idea but only in case "Toggle" - 1 command for "on" and "off". 
And not for all  Gms. Only for you. Think about it.

**Pull request closed without merging**